Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -155,802 +155,844 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
     return GetDefaultReply(IPC_EACCES);
   }
 
-  s32 return_value = 0;
   switch (request.request)
   {
   case IOCTL_SO_STARTUP:
-  {
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
-    break;
-  }
+    return StartUp(request);
   case IOCTL_SO_SOCKET:
-  {
-    u32 af = Memory::Read_U32(request.buffer_in);
-    u32 type = Memory::Read_U32(request.buffer_in + 4);
-    u32 prot = Memory::Read_U32(request.buffer_in + 8);
-
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.NewSocket(af, type, prot);
-    INFO_LOG(IOS_NET, "IOCTL_SO_SOCKET "
-                      "Socket: %08x (%d,%d,%d), BufferIn: (%08x, %i), BufferOut: (%08x, %i)",
-             return_value, af, type, prot, request.buffer_in, request.buffer_in_size,
-             request.buffer_out, request.buffer_out_size);
-    break;
-  }
+    return Socket(request);
   case IOCTL_SO_ICMPSOCKET:
-  {
-    u32 pf = Memory::Read_U32(request.buffer_in);
-
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.NewSocket(pf, SOCK_RAW, IPPROTO_ICMP);
-    INFO_LOG(IOS_NET, "IOCTL_SO_ICMPSOCKET(%x) %d", pf, return_value);
-    break;
-  }
+    return ICMPSocket(request);
   case IOCTL_SO_CLOSE:
   case IOCTL_SO_ICMPCLOSE:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    return_value = sm.DeleteSocket(fd);
-    INFO_LOG(IOS_NET, "%s(%x) %x",
-             request.request == IOCTL_SO_ICMPCLOSE ? "IOCTL_SO_ICMPCLOSE" : "IOCTL_SO_CLOSE", fd,
-             return_value);
-    break;
-  }
+    return Close(request);
   case IOCTL_SO_ACCEPT:
   case IOCTL_SO_BIND:
   case IOCTL_SO_CONNECT:
   case IOCTL_SO_FCNTL:
-  {
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    WiiSockMan& sm = WiiSockMan::GetInstance();
-    sm.DoSock(fd, request, static_cast<NET_IOCTL>(request.request));
-    return GetNoReply();
-  }
-  /////////////////////////////////////////////////////////////
-  //                  TODO: Tidy all below                   //
-  /////////////////////////////////////////////////////////////
+    return DoSock(request);
   case IOCTL_SO_SHUTDOWN:
-  {
-    request.Log(GetDeviceName(), LogTypes::IOS_WC24);
-
-    u32 fd = Memory::Read_U32(request.buffer_in);
-    u32 how = Memory::Read_U32(request.buffer_in + 4);
-    int ret = shutdown(fd, how);
-    return_value = WiiSockMan::GetNetErrorCode(ret, "SO_SHUTDOWN", false);
+    return Shutdown(request);
+  case IOCTL_SO_LISTEN:
+    return Listen(request);
+  case IOCTL_SO_GETSOCKOPT:
+    return GetSockOpt(request);
+  case IOCTL_SO_SETSOCKOPT:
+    return SetSockOpt(request);
+  case IOCTL_SO_GETSOCKNAME:
+    return GetSockName(request);
+  case IOCTL_SO_GETPEERNAME:
+    return GetPeerName(request);
+  case IOCTL_SO_GETHOSTID:
+    return GetHostID(request);
+  case IOCTL_SO_INETATON:
+    return INetAToN(request);
```

Comment:
Nitpicking, but the capitalisation looks a bit weird here. Could this be InetAToN instead?